File: common/src/main/java/cc/turtl/chiselmon/system/dashboard/api/endpoints/ContextHandler.java
package cc.turtl.chiselmon.system.dashboard.api.endpoints;

import cc.turtl.chiselmon.ChiselmonConstants;
import cc.turtl.chiselmon.system.dashboard.api.ApiHandler;
import cc.turtl.chiselmon.system.tracker.EncounterDatabase;
import com.sun.net.httpserver.HttpExchange;

import java.io.IOException;

public class ContextHandler extends ApiHandler {

    private record ContextResponse(String worldType, String worldName, String modVersion) {
    }

    public ContextHandler(EncounterDatabase db) {
        super(db);
    }

    @Override
    public void handle(HttpExchange exchange) throws IOException {
        handleRequest(exchange, (timeRange, params) -> {
            String worldType, worldName;
            String dbFolder = db.getDbPath().getParent().getFileName().toString();

            if (dbFolder.startsWith("mp-")) {
                worldType = "mp";
                worldName = dbFolder.substring(3);
            } else if (dbFolder.startsWith("sp-")) {
                worldType = "sp";
                worldName = dbFolder.substring(3).replace('_', ' ');
            } else {
                worldType = "unknown";
                worldName = "unknown";
            }

            return new ContextResponse(worldType, worldName, ChiselmonConstants.VERSION);
        });
    }
}

File: common/src/main/kotlin/cc/turtl/chiselmon/client/system/dashboard/api/endpoints/ContextHandler.kt
package cc.turtl.chiselmon.client.system.dashboard.api.endpoints

import cc.turtl.chiselmon.BuildDetails
import cc.turtl.chiselmon.client.system.dashboard.api.ApiHandler
import cc.turtl.chiselmon.client.system.tracker.EncounterDatabase
import com.sun.net.httpserver.HttpExchange

class ContextHandler(db: EncounterDatabase) : ApiHandler(db) {

    private data class ContextResponse(val worldType: String, val worldName: String, val modVersion: String)

    override fun handle(exchange: HttpExchange) {
        handleRequest(exchange) { _, _ ->
            val dbFolder = db.dbPath.parent.fileName.toString()
            val (worldType, worldName) = when {
                dbFolder.startsWith("mp-") -> "mp" to dbFolder.substring(3)
                dbFolder.startsWith("sp-") -> "sp" to dbFolder.substring(3).replace('_', ' ')
                else -> "unknown" to "unknown"
            }
            ContextResponse(worldType, worldName, BuildDetails.MOD_VERSION)
        }
    }
}
